v2/tools/generator/internal/conversions/property_conversions.go

```diff
@@ -1171,6 +1172,105 @@ func assignMapFromMap(
 	}, nil
 }
 
+// assignUserAssignedIdentityMapFromArray will generate a code fragment to populate a userAssignedIdentity array from
+// a map whose key is the ARM ID of the userAssignedIdentity
+// <arr> := make([]<arrType>, 0, len(source.UserAssignedIdentities))
+//
+//	for key, _ := range source.UserAssignedIdentities {
+//	   ref := genruntime.CreateResourceReferenceFromARMID(key)
+//	   <arr> = append(<arr>, UserAssignedIdentityDetails{Reference: ref})
+//	}
+//
+// <writer> = <arr>
+func assignUserAssignedIdentityMapFromArray(
+	sourceEndpoint *TypedConversionEndpoint,
+	destinationEndpoint *TypedConversionEndpoint,
+	conversionContext *PropertyConversionContext) (PropertyConversion, error) {
+
+	// Require both source and destination to not be bag items
+	if sourceEndpoint.IsBagItem() || destinationEndpoint.IsBagItem() {
+		return nil, nil
+	}
+
+	// Require source to be a map type
+	_, sourceIsMap := astmodel.AsMapType(sourceEndpoint.Type())
+	if !sourceIsMap {
+		return nil, nil
+	}
+
+	// Require destination to be an array type
+	destinationArray, destinationIsArray := astmodel.AsArrayType(destinationEndpoint.Type())
+	if !destinationIsArray {
+		return nil, nil
+	}
+
+	if sourceEndpoint.Name() != astmodel.UserAssignedIdentitiesProperty {
+		return nil, nil
+	}
+
+	if destinationEndpoint.Name() != astmodel.UserAssignedIdentitiesProperty {
+		return nil, nil
+	}
+
+	destinationElement := destinationArray.Element()
+
+	conversion, err := CreateTypeConversion(
+		sourceEndpoint.WithType(astmodel.StringType),
+		destinationEndpoint.WithType(astmodel.ResourceReferenceType),
+		conversionContext)
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"finding UserAssignedIdentities conversion from %s to %s",
+			astmodel.DebugDescription(astmodel.StringType),
+			astmodel.DebugDescription(astmodel.ResourceReferenceType))
+	}
+
+	return func(reader dst.Expr, writer func(dst.Expr) []dst.Stmt, knownLocals *astmodel.KnownLocalsSet, generationContext *astmodel.CodeGenerationContext) []dst.Stmt {
```

Comment:
I'd suggest annotating individual statements with comments showing what code will be generated.

Reply:
Added these